_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,241 @@
+---
+layout: post
+title: "Hacking Strimzi for Cruise Control UI"
+date: 2022-10-31
+author: kyle_liberti
+---
+
+For some time now, Strimzi has offered Kubernetes native support Cruise Control allowing users to safely configure and communicate with Cruise Control through Strimzi custom resources.
+This is great but sometimes it is a better user experience to interact with applications like Cruise Control through a UI.
+Although Strimzi does not offer direct support for the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) it can still be run alongside and connected to a Strimzi managed Cruise Control instance.
+
+This post will walk through how to do so! 
+
+## Prerequisites
+
+For starters we will need a Strimzi managed Kafka cluster with Cruise Control enabled.  
+
+```
+                           *  *
+                        *        *
+                       *  Cruise  *
+                       *  Control *
+                        *        *
+                           *  *
+                            ^                   *  *         *  *          *  *
+                            |                *        *   *        *    *        *
++----------+                |               *  Kafka   * *  Kafka   *  *  Kafka   *
+|          |               *  *             *   Pod    * *   Pod    *  *   Pod    *
+|  Kafka   |            *        *           *        *   *        *    *        *
+| Resource | <------+  * Cluster  * ----->      *  *         *  *          *  *
+|          | +------>  * Operator *
+|          |            *        *              *  *         *  *          *  *
++----------+               *  *              *        *   *        *    *        *
+                                            * Zookeeper* * Zookeeper*  * Zookeeper*
+                                            *   Pod    * *   Pod    *  *   Pod    *
+                                             *        *   *        *    *        *
+                                                *  *         *  *          *  *
+
+Kubernetes Land                                          Kafka Cluster
+```
+Since the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) does not support HTTPs and Strimzi does not support creating custom Cruise Control API user roles we will need to disable Cruise Control SSL and HTTP basic authentication settings. 
```

Comment:
- `does not support HTTPs` Is this true? When installed in the same container as Cruise Control the UI reuses the Cruise Control REST API setup so it should work with HTTPS. I think it should be possible to still use HTTPS with the proxy approach you took.
- `HTTPs` -> `HTTPS`
- Do we need the Kafka and ZooKeeper pods in the diagram? What about only having `Kafka Cluster`? That way the diagram would also work for KRaft clusters.